modules/custom/az_cas/src/Service/GuestSessionManager.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+namespace Drupal\az_cas\Service;
+
+use Drupal\Component\Datetime\TimeInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\State\StateInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Service for managing guest authentication sessions.
+ */
+class GuestSessionManager {
```

Comment:
Does this class potentially reinvent some functionality that's already available in the session service?